tdash/generators.py

```diff
@@ -8,3 +8,8 @@ def flatten(gen):
     for arr in gen:
         for elem in arrays.flatten(arr, is_deep=True):
             yield elem
+
+def split_list(self, input_data, n):
+    """ Yield successive n-sized chunks from input_data """
+    for i in xrange(0, len(input_data), n):
+        yield input_data[i:i+n]
```

Comment:
does this work ok when i+n > len(input_data) ?

This will not work if input_data is a generator because you use array access. Change to an implementation that works if input_data is a generator - it's more generic.